webpack.config.js

```diff
@@ -1,56 +1,71 @@
 const webpack = require('webpack');
 const path = require('path');
 const CopyWebpackPlugin = require('copy-webpack-plugin');
-const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 
 module.exports = {
-  devtool: 'cheap-source-map',
+  mode: 'production',
+  devtool: 'source-map',
   entry: [
-    path.resolve(__dirname, 'app/main.js'),
+    path.resolve(__dirname, 'app/main.tsx'),
     path.resolve(__dirname, 'app/stylesheets/main.scss'),
   ],
   output: {
     path: path.resolve(__dirname, 'dist'),
     publicPath: '/',
-    filename: './dist.js'
+    filename: './dist.js',
   },
   module: {
     rules: [
-      { test: /\.css$/, include: path.resolve(__dirname, 'app'), loader: 'style-loader!css-loader' },
       {
-        test: /\.scss$/,
+        test: /\.s?css$/,
+        include: path.resolve(__dirname, 'app'),
+        loader: [MiniCssExtractPlugin.loader, 'css-loader', 'sass-loader'],
+      },
+      {
+        test: /\.ts(x?)$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'ts-loader',
+          },
+        ],
+      },
+      // All output '.js' files will have any sourcemaps re-processed by 'source-map-loader'.
+      {
+        enforce: 'pre',
+        test: /\.js$/,
         exclude: /node_modules/,
-        use: ExtractTextPlugin.extract({
-          fallback: 'style-loader',
-          use: [
-            'css-loader',
-            { loader: 'sass-loader', query: { sourceMap: false } },
-          ],
-          publicPath: '../'
-        }),
+        loader: 'source-map-loader',
       },
-      { test: /\.js[x]?$/, include: [
-        path.resolve(__dirname, 'app'),
-        path.resolve(__dirname, 'node_modules/sn-components-api/dist/dist.js')
-      ], exclude: /node_modules/, loader: 'babel-loader' }
-    ]
+    ],
+  },
+  externals: {
+    'filesafe-js': 'filesafe-js',
   },
   resolve: {
-    extensions: ['.js', '.jsx', '.css', '.scss'],
+    modules: ['node_modules', path.resolve(__dirname, 'app')],
+    extensions: ['.ts', '.tsx', '.js', '.jsx', '.css', '.scss'],
     alias: {
-        highlightjs_css: path.join(__dirname, 'node_modules/highlight.js/styles/atom-one-light.css'),
-        stylekit: path.join(__dirname, 'node_modules/sn-stylekit/dist/stylekit.css')
-    }
+      highlightjs_css: path.join(
+        __dirname,
+        'node_modules/highlight.js/styles/atom-one-light.css'
+      ),
+      stylekit: path.join(
+        __dirname,
+        'node_modules/sn-stylekit/dist/stylekit.css'
+      ),
+    },
   },
   plugins: [
-    new ExtractTextPlugin({ filename: './dist.css', disable: false, allChunks: true}),
+    new MiniCssExtractPlugin({
+      filename: 'dist.css',
+    }),
     new webpack.DefinePlugin({
```

Comment:
I was updating the Bold Editor webpack config, I think that this is not needed in the later versions.